### PR TITLE
wiki docs generation - fixups and fork ops

### PR DESCRIPTION
**Pull request by briantist**
With #21 landed, found a few issues:
- if fails on forks (unless the wiki is enabled, and the `actions/` and `workflows/` directories existed in the repo)
- the setting I was using for `GH_NAME` was wrong; it was supposed to be a username.

So this PR makes the following changes:
- we set an env var `SHOULD_RUN` that checks whether we're running in the main repo, or whether a secret named `GENERATE_WIKI_PAGES` is set to `yes`
- all steps in the job are now conditional, and only run if `SHOULD_RUN` is true
- the Ansible role to generate the docs ensures the target directory exists
- added `.internal/ansible/**` to the list of paths that trigger the workflow
- [minor] moved adding the trailing forward slash on the wiki dir to the action invocation

So, on **forks**, this workflow will no longer fail. If you _want_ it to run on your fork, you must enable the wiki, and you must set the secret in your fork.

Without the above, all of the job's steps will be skipped, however the job will still run (allocate a VM, build the action container, etc.). This is because we cannot use the `secrets` context nor the `env` context in the `jobs.<job_id>.if` conditional,s o there's no way we can use the secret trigger method AND skip the job as a whole.

---

Further thoughts: we might have more workflows (certain types of tests and other repo-level stuff, like maybe what @felixfontein proposed in #4 ), that should be opt-in or not run at all on forks. One way we can avoid putting the conditionals on every step, and instead use job-level conditionals, is to run a separate job that figures out the values, and sets outputs.

The jobs that need the information could use `needs:` to wait on that first job. I really don't like that pattern, but it's one of the only ways to get around some of the more frustrating limitations in GHA.

If we did that, we might consider a shared workflow for that initial fork-checking job, that way it can be re-used by this and future workflows.

Perhaps I will wait until the _second_ workflow that needs this pattern before implementing this...

### CODE REVIEW

**Review by felixfontein — APPROVED** [review]

File: .internal/ansible/ansible_collections/internal/gha_docs/roles/generate/tasks/main.yml [hunk 1]
```diff
@@ -3,6 +3,12 @@
   set_fact:
     _src: "{{ lookup('ansible.builtin.file', file) | from_yaml }}"
 
+- name: Ensure output dir exists
+  ansible.builtin.file:
+    path: '{{ output | dirname }}'
+    state: directory
+    mode: '744'
```
Comment (felixfontein) [inline .internal/ansible/ansible_collections/internal/gha_docs/roles/generate/tasks/main.yml:10]: Ah, one thing: shouldn't this be `755`? Or is it intentional that the executable bit is missing, while reading is allowed?
```suggestion
    mode: '755'
```

Comment (briantist) [reply .internal/ansible/ansible_collections/internal/gha_docs/roles/generate/tasks/main.yml:10]: I would think we don't need the executable bit, the directories will contain markdown files. Thought it's entire;y possible I'm missing something...

Comment (felixfontein) [reply .internal/ansible/ansible_collections/internal/gha_docs/roles/generate/tasks/main.yml:10]: You cannot access any files in the folder if the folder has no executable bit. Basically only `700` or `755` make sense.

Comment (briantist) [reply .internal/ansible/ansible_collections/internal/gha_docs/roles/generate/tasks/main.yml:10]: Ah, I forgot about the meaning of X bit in dirs. 744 worked fine in testing (within the workflow) but perhaps locally that would be problematic.

**Review by felixfontein — APPROVED** [review]

File: .internal/ansible/ansible_collections/internal/gha_docs/roles/generate/tasks/main.yml (hunk 1 above)
Comment (briantist) [reply .internal/ansible/ansible_collections/internal/gha_docs/roles/generate/tasks/main.yml]: thank you @felixfontein good catch, and thanks for reviewing again!

**Pull request merged**